### PR TITLE
Add types to toJSON and actionToJSON methods

### PR DESCRIPTION
**Pull request by cubismic**
I added types to toJSON and actionToJSON methods.
Except for src/core/Invoice.ts.

Please review for the Buffer in the JSON type. I can't find any better way to replace those.

### CODE REVIEW

File: src/core/Block.ts [hunk 1]
```diff
@@ -20,6 +21,20 @@ export interface BlockData {
     hash: H256;
     transactions: SignedTransaction[];
 }
+export interface BlockJSON {
+    parentHash: string;
+    timestamp: number;
+    number: number;
+    author: string;
+    extraData: Buffer;
```
Comment (joojis) [inline src/core/Block.ts]: number[]

File: src/core/Block.ts [hunk 2]
```diff
@@ -20,6 +21,20 @@ export interface BlockData {
...
+    transactionsRoot: string;
+    stateRoot: string;
+    invoicesRoot: string;
+    score: string;
+    seal: Buffer[];
```
Comment (joojis) [inline src/core/Block.ts]: number[][]

File: src/core/transaction/Custom.ts [hunk 3]
```diff
@@ -3,6 +3,12 @@ import { U64 } from "codechain-primitives";
 import { Transaction } from "../Transaction";
 import { NetworkId } from "../types";
 
+export interface CustomActionJSON {
+    handlerId: string;
+    buffer: Buffer;
```
Comment (joojis) [inline src/core/transaction/Custom.ts]: number[]

File: src/core/transaction/CreateShard.ts [hunk 4]
```diff
@@ -1,6 +1,8 @@
 import { Transaction } from "../Transaction";
 import { NetworkId } from "../types";
 
+export type CreateShardActionJSON = any;
```
Comment (joojis) [inline src/core/transaction/CreateShard.ts]: {}

File: src/core/transaction/Pay.ts [hunk 5]
```diff
@@ -2,6 +2,11 @@ import { PlatformAddress, U64 } from "../classes";
 import { Transaction } from "../Transaction";
 import { NetworkId } from "../types";
 
+export interface PayActionJSON {
+    receiver: string;
+    quantity: string | number;
```
Comment (joojis) [inline src/core/transaction/Pay.ts]: The type of quantity is string

File: src/core/transaction/CreateShard.ts (hunk 4 above)
Comment (joojis) [reply src/core/transaction/CreateShard.ts]: https://palantir.github.io/tslint/usage/rule-flags/

**Comment by cubismic** [comment]
I've changed your requests!

File: src/core/SignedTransaction.ts [hunk 6]
```diff
@@ -124,7 +124,7 @@ export class SignedTransaction {
      * Convert to a SignedTransaction JSON object.
      * @returns A SignedTransaction JSON object.
      */
-    public toJSON() {
+    public toJSON(): TransactionJSON {
```
Comment (joojis) [inline src/core/SignedTransaction.ts]: You need to add SignedTransactionJSON and change it.

**Pull request merged**